client/ladder/ladder.tsx

```diff
@@ -120,38 +130,54 @@ export function LadderRouteComponent(props: { params: any }) {
   const matchmakingType = ALL_MATCHMAKING_TYPES.includes(params.matchmakingType as MatchmakingType)
     ? (params.matchmakingType as MatchmakingType)
     : undefined
+  const seasonId = params.seasonId ? makeSeasonId(Number(params.seasonId)) : undefined
 
-  return <Ladder matchmakingType={matchmakingType} />
+  return <Ladder matchmakingType={matchmakingType} seasonId={seasonId} />
 }
 
 export interface LadderProps {
   matchmakingType?: MatchmakingType
+  seasonId?: SeasonId
 }
 
 /**
  * Displays a ranked table of players on the ladder(s).
  */
-export function Ladder({ matchmakingType: routeType }: LadderProps) {
+export function Ladder({ matchmakingType: routeType, seasonId }: LadderProps) {
   const matchmakingType = routeType ?? savedLadderTab.getValue() ?? MatchmakingType.Match1v1
   useTrackPageView(urlPath`/ladder/${matchmakingType}`)
   const { t } = useTranslation()
   const dispatch = useAppDispatch()
-  const rankings = useAppSelector(s => s.ladder.typeToRankings.get(matchmakingType))
-  const searchResults = useAppSelector(s => s.ladder.typeToSearchResults.get(matchmakingType))
-  const usersById = useAppSelector(s => s.users.byId)
-
-  const rankingsSeason = useAppSelector(s =>
-    rankings?.seasonId ? s.matchmakingSeasons.byId.get(rankings.seasonId) : undefined,
+  const seasons = useAppSelector(s => s.matchmakingSeasons.byId)
+  const currentSeasonId = useAppSelector(s => s.matchmakingSeasons.currentSeasonId)
```

Comment:
If you have the current season ID here, does the reducer even need a separate non-season-based storage of any of this? Seems useless and makes all the logic after this far more complex when you could just retrieve the same data from the same place always

Reply:
Yeah, I was debating this with myself as well. We _don't_ have the current season ID here actually, we fetch it in the effect below which fetches all the matchmaking seasons (and returns the ID of the current season). But you're right, it seems like a whole bunch of stuff would be simplified if we simply fetched the matchmaking seasons (and the current season ID) in the component above this one and show a loader (or nothing at all) until that request is finished?